addons/point_of_sale/wizard/pos_make_invoice.py

```diff
@@ -0,0 +1,73 @@
+import pytz
+
+from odoo import api, models, fields, Command, _
+from odoo.exceptions import UserError
+from collections import defaultdict
+
+
+class PosMakeInvoice(models.TransientModel):
+    _name = 'pos.make.invoice'
+    _description = 'Multiple order invoice creation'
+
+    consolidated_billing = fields.Boolean(
+        string="Consolidated Billing", default=True,
+        help="Create one invoice for all orders related to same customer and same invoicing address"
+    )
+    count = fields.Integer(string="Order Count", compute='_compute_order_count')
+
+    @api.depends('consolidated_billing')
+    def _compute_order_count(self):
+        for wizard in self:
+            wizard.count = len(self.env.context.get('active_ids'))
+
+    def action_create_invoices(self):
+        self.ensure_one()
+        selected_orders = self.env['pos.order'].browse(self.env.context.get('active_ids'))
+        # Checking requirements
+        if not (all([o.partner_id for o in selected_orders]) and len(selected_orders.mapped('partner_id')) == 1):
+            partner = selected_orders.partner_id
+            if len(partner) == 1:
+                return {
+                    'name': _('Warning'),
+                    'view_mode': 'form',
+                    'view_id': self.env.ref('point_of_sale.view_confirm_action_wizard').id,
+                    'res_model': 'pos.confirmation.wizard',
+                    'target': 'new',
+                    'type': 'ir.actions.act_window',
+                    'context': {'orders': selected_orders.ids, 'dialog_size': 'medium'},
+                }
+            else:
+                raise UserError(_("You should select at least one order with a partner."))
```

Comment:
also take into account the situation where the selected orders points to multiple partners.